File: Frontend/src/infrastructure/dtos/operational-window/OperationalWindowFormDto.ts
import { DAYS_OF_WEEK } from "../../../domain/Enums/daysOfWeek";

export type DayOfWeek = (typeof DAYS_OF_WEEK)[number];

export type operationalWindowFormDto = {
  code: string;
  dayOfWeek: DayOfWeek;
  startHour: number;
  startMinute: number;
  endHour: number;
  endMinute: number;
};

File: JWP_API/JadeWesserPort/DTOs/OperationalWindowDTOs/OperationalWindowFormDTO.cs
namespace JadeWesserPort.DTOs.OperationalWindowDTOs;

public class OperationalWindowFormDTO
{
    public string Code { get; set; } = string.Empty;    
    public DayOfWeek DayOfWeek { get; set; }
    public int StartHour { get; set; }
    public int StartMinute { get; set; }
    public int EndHour { get; set; }
    public int EndMinute { get; set; }


    public TimeOnly StartTime => new(StartHour, StartMinute);
    public TimeOnly EndTime => new(EndHour, EndMinute);
}